ehrql/backends/base.py

```diff
@@ -44,23 +46,22 @@ def _init_table(cls, table):
         """
         table.learn_patient_join(cls.patient_join_column)
 
-    @classmethod
-    def validate_against_table_namespace(cls, table_namespace):
+    def validate_against_table_namespace(self, table_namespace):
         for attr, ql_table in get_tables_from_namespace(table_namespace):
             table = ql_table.qm_node
-            if table.name not in cls.tables:
+            if table.name not in self.tables:
                 raise ValidationError(
-                    f"{cls} does not implement table '{table.name}' from "
+                    f"{self.__class__} does not implement table '{table.name}' from "
                     f"{table_namespace}.{attr}"
                 )
 
-            table_def = cls.tables[table.name]
+            table_def = self.tables[table.name]
             try:
-                table_def.validate_against_table_schema(table.schema)
+                table_def.validate_against_table_schema(self, table.schema)
```

Comment:
I was puzzled by `self` here - I thought this `self` would replace the `table_def.validate_against_table_schema(self, ...)` (possibly a half-remembered piece of JavaScript?). Of course, it doesn't: this `self` is an instance of `BaseBackend`, which is the second positional argument to `MappedTable.validate_against_table_schema`.

To my inexperienced eye, passing `self` like this looks unusual. Instead, how about:

```suggestion
                table_def.validate_against_table_schema(backend=self, schema=table.schema)
```

Reply:
Good suggestion, thanks.